modular_zubbers/code/modules/antagonists/malf/infected_ipc.dm

```diff
@@ -0,0 +1,155 @@
+/datum/antagonist/infected_ipc
+	name = "\improper Infected Synthetic"
+	roundend_category = "traitors"
+	antagpanel_category = "Malf AI"
+	job_rank = ROLE_INFECTED_SYNTHETIC
+	antag_hud_name = "traitor"
+	ui_name = "AntagInfoHackedSynthetic"
+	antag_moodlet = /datum/mood_event/infected_ipc
+	///radio binary for synthetic
+	var/obj/item/implant/radio/infected_ipc/internal_radio
+	/// The camera for ai
+	var/obj/machinery/camera/internal_camera
+	///ai that's linked to the synthetic
+	var/mob/living/silicon/ai/master_ai
+
+/datum/antagonist/infected_ipc/admin_add(datum/mind/new_owner, mob/admin)
+	var/mob/living/carbon/target = new_owner.current
+	var/mob/living/silicon/ai/owner_ai
+	if(!istype(target))
+		to_chat(admin, "Infected Synthetics come from a brain trauma, so they need to at least be a carbon!")
+		return
+	if(!target.get_organ_by_type(/obj/item/organ/brain))
+		to_chat(admin, "Infected Synthetics come from a brain trauma, so they need to HAVE A BRAIN.")
+		return
+	var/chosen = tgui_input_list(admin, "Pick AI for the Synthetic to be bound to:", "Pick AI", GLOB.ai_list)
```

Comment:
```suggestion
	var/chosen = tgui_input_list(admin, "Pick AI for the Synthetic to be bound to:", "Pick AI", GLOB.ai_list)
	if(!chosen)
		return
```
TGUI inputs return null if the user cancels them so you always wanna have a case for this